### ISSUE TITLE
'adb install any.apk' not working with my Nokia 6.1 (PL2) w/ crDroid 10.2 [OFFICIAL] connected to my (Intel) Mac

### ISSUE THREAD

**Issue by LeeBinder** [crdroidandroid/issue_tracker#343]
### Issue type

Feature not working as intended

### Device

Nokia 6.1 (PL2)

### crDroid version

crDroid 10

### Exact version / Build date

10.2 - any build from https://crdroid.net/PL2/10, latest 2024-03-18 to oldest 2024-02-26

### Bug description

On my Intel Mac in both, macOS Sonoma and Monterey, after enabling dev options then USB debugging and confirming the authentication request on the phone, when trying to install any apk via:

`adb install path/to/some/android-app.apk` or (with re-install switch) `adb install -r path/to/some/android-app.apk`

the process in Terminal gets stuck at

> Performing Streamed Install

I tried several apks, all of which I know are compatible. This was still working fine with the [unofficial crDroid 10.0 for my Nokia 6.1](https://xdaforums.com/t/rom-unofficial-14-pl2-crdroid-10-vanilla.4645786/).

Also, when I connect my Samsung, the exact same command runs through with "Success" with each and every apk. I've never experienced this issue before with any other ROMs I installed onto my Nokia. After downgrading to crDroid 10.0, `adb install any.apk` was working again right away. Same applies to the latest and any older official LineageOS for my device.

Latest adb 35.0.0-11411520 (1.0.41); issue persists regardless if adb installed from Google or via homebrew. Also, `adb devices` correctly issues

>     List of devices attached
>     PL2GAR9842318434 device

Also tried the last 5 adb versions - same thing.

All other adb commands I tested work. Apart from rebooting both, my phone and my Mac, using all possible USB ports and several cables, trying the install command in a different user account, I've also tried all the hints and tips from [here](https://stackoverflow.com/questions/23081263/adb-android-device-unauthorized/25546300#25546300) and [there](https://stackoverflow.com/questions/31638582/android-adb-devices-unauthorized), to no avail.

The OS matters, because on Windows it's fine, just not working in macOS with my Nokia 6.1 + [ROM][14][OFFICIAL] crDroid Android v10.2 [PL2] ([crdroid.net](https://crdroid.net/PL2/10) | [XDA](https://xdaforums.com/t/rom-14-official-crdroid-android-v10-2-pl2.4657869/)).

Summed up, this is definitely a bug either of crDroid 10.2 in general, or of this particular crDroid 10.2 for Nokia 6.1 build. I've posted into both, the [XDA-forum topic](https://xdaforums.com/t/rom-14-official-crdroid-android-v10-2-pl2.4657869/post-89403184), the Nokia 6.1 PL2 crDroid Telegram group and the general crDroid group, but haven't received any help or even feedback yet.

What I kindly request is:

a) someone with a Mac and a device with crDroid 10.2, can you please test and report in here if you can reproduce this or not? Please in any case indicate your device
b) Is there anything I can patch?

Thank you for looking into this.

[logcat.txz.zip](https://github.com/crdroidandroid/issue_tracker/files/14673507/logcat.txz.zip)

### Steps to reproduce

Mobile device:
1.  enable dev options > USB debugging
2. connect your device to your Mac via USB > confirm authentication

Mac:
1. install adb
3. in Terminal: `adb devices && adb install -r path/to/some/android-app.apk` (adapt the path)

Does it get stuck at "Performing Streamed Install", or finish with "Success"?

### Relevant log (logcat/build log)

```shell
[logcat.txz.zip](https://github.com/crdroidandroid/issue_tracker/files/14673507/logcat.txz.zip)
```

### Acknowledgements

- [X] I've checked device is officially supported (for device specific reports and not source related).
- [X] I'm running latest version available on crdroid.net for this device
- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by gwolf2u** [comment]
unfortunately this is a really specific use case for you
did check logcat but can't say I see anything specific
also if you say it works on Windows and not on MAC, well I incline to say is MAC driver issue and not crDroid
some stuff change between revisions of crDroid (Google side) that can render a OS driver obsolete or incompatible with some features, so OS will need to update those drivers for compatibility
reason you say it worked in previous update and not latest

so due to above reasons, I will have to close this ticket for now

**Comment by LeeBinder** [comment]
Hi Lup Gabriel. crDroid is based on LineageOS. My phone's dev who also has a Mac confirmed two days ago that LOS, not even his very latest build, is affected by this bug, so it's clearly a crDroid 10.2 and NOT a Mac adb driver bug. We really may not expect the Android Platform Tools devs to mitigate a quirk some custom ROM introduces during some dev process - that would be a 180° reality twist. And that Windows hasn't been affected all the while because even crDroid 10.2 continues to allow Windows adb access as supposed is just a side note and thus irrelevant.

With this topic now closed, nobody will find it easily if at all and thus nobody can contribute by adding if they, too, are affected, or not.

A topic should be closed if:
- it has been solved
- or has turned obsolete
- or simply can't be solved

None of the above applies.

As long as we don't have more data from others, esp. to differentiate if this is a general crDroid 10.2 or a Nokia 6.1 (PL2) only crDroid 10.2 Mac adb issue, I do but kindly request you to please re-open this issue and keep it open until we have gained at least that differentiation, or at least for one more week. Thank you.

Re. logcat: I ran `adb logcat -d -b main -b system -b events -v time > logcat.txt`. I sense this might not be sufficient. Can you please tell me a better adb relevant logcat inquiry, if you know it?

**Comment by LeeBinder** [comment]
I just installed the very latest adb v.35.0.1-11580240 update. Needless to say that didn't fix this crDroid 10.2 adb issue.

**Comment by LeeBinder** [comment]
@gwolf2u I must've made a mistake late at nite when I was debugging this before I opened this issue, because this bug also occurs in Windows with the phone connected to the computer, also even with the latest adb. In hindsight I think what happened was that I had already downgraded back to (or had simply switched back to the other slot with) crDroid 10.0 before testing this in Windows.

I'll now check in with other crDroid 10.2 users regardless of their computer's OS and take it from there. Also, Lup, since you most likely are running crDroid 10.2 (or pre-10.3) and this indeed is computer-OS independent, would you please let me know your result of `adb devices && adb install -r path/to/some/android-app.apk` (adapt the path)

Does it get stuck at "Performing Streamed Install", or finish with "Success"?

**Comment by LeeBinder** [comment]
matlog adb & package manager related digest:

```javascript
03-28 22:58:34.973 E/ResilientAtomicFile(1952): Failed to verity-protect package restrictions
03-28 22:58:34.973 E/ResilientAtomicFile(1952): java.io.IOException: Failed to enable fs-verity on FD(391): Inappropriate ioctl for device
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.internal.security.VerityUtils.setUpFsverity(VerityUtils.java:89)
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.server.security.FileIntegrity.setUpFsVerity(FileIntegrity.java:56)
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.server.pm.ResilientAtomicFile.finishWrite(ResilientAtomicFile.java:146)
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.server.pm.Settings.writePackageRestrictions(Settings.java:2336)
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.server.pm.Settings.writePackageRestrictions(Settings.java:2171)
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.server.pm.PackageManagerService.writePendingRestrictions(PackageManagerService.java:1483)
03-28 22:58:34.973 E/ResilientAtomicFile(1952): 	at com.android.server.pm.PackageManagerService$2.handleMessage(PackageManagerService.java:1510)

03-28 22:58:34.997 I/adbd    (1443): UsbFfs: offline
03-28 22:58:34.997 I/adbd    (1443): Calling send_auth_request...
03-28 22:58:35.005 I/adbd    (1443): adbd_auth: loading keys from /data/misc/adb/adb_keys
03-28 22:58:35.008 I/adbd    (1443): adb client authorized

03-28 22:59:20.406 W/adbd    (1443): timeout expired while flushing socket, closing
```

**Comment by LeeBinder** [comment]
Working = solved with the 10.4 update 👍
